src/utils/toasts/customToastsActions.tsx

```diff
@@ -0,0 +1,33 @@
+import { addCustomToast, removeCustomToast } from 'reduxStore/slices';
+import { store } from 'reduxStore/store';
+import { CustomToastType } from '../../types/toasts.types';
+
+/**
+ * React components cannot be sent trough Custom Events nor trough Redux
+ * So a local dictionary is created here for registerd components
+ */
+const componentsDictionary: Record<string, () => JSX.Element> = {};
+
+export const getRegisteredCustomIconComponents = (id: string) =>
+  componentsDictionary[id];
+
+export const addNewCustomToast = (props: CustomToastType) => {
+  if (props.component != null) {
+    const { component, ...rest } = props;
+    const { payload } = store.dispatch(
+      addCustomToast({
+        ...rest,
+        // do not send component to Redux
+        component: null
+      })
+    );
+
+    componentsDictionary[payload.toastId] = component;
```

Comment:
just to make sure, you'd probably want to check here that component is != null

Reply:
this is assignment so no need to check for null. overwriting I think is OK